src/routes/articles.route.js

```diff
@@ -253,11 +257,48 @@ articleRouter.post('/:articleId/like', checkAuthorizedUser, likeAnArticle);
 /**
  * @swagger
  *
- * /api/v1/article/:articleId/unlike:
+ * /api/v1/article/:articleId/rate:
  *   post:
  *     tags:
  *       - article
- *     description: users can like an article.
+ *     description: users can rate a single article.
+ *     produces:
+ *       - application/json
+ *     parameters:
+ *       - name: articleId
+ *         description: the id of the article.
+ *         in: params
+ *         required: true
+ *     request:
+ *         content:
+ *         - application/json
+ *         schema:
+ *           type: array
+ *           items:
+ *         $ref: '#/definitions/article'
+ *     responses:
+ *       200:
+ *         description: article rated
+ *       404:
+ *         description: article not found
+ *       500:
+ *         description: Database error
+ */
+articleRouter.post(
+  '/:articleId/rate',
+  checkAuthorizedUser,
+  validateRating,
```

Comment:
I think we should validate the request body first before we proceed to hit the database. i.e. `validateRating` before `checkAuthorizedUser`